plasmapy/diagnostics/tests/test_thomson.py

```diff
@@ -0,0 +1,251 @@
+
+def gen_collective_spectrum():
+    """
+    Generates an example Thomson scattering spectrum in the collective regime
+    """
+    wavelengths = np.arange(520, 545, 0.01) * u.nm
+    probe_wavelength = 532 * u.nm
+    ne = 5e17 * u.cm ** -3
+    probe_vec = np.array([1, 0, 0])
+    scatter_vec = np.array([0, 1, 0])
+    fract = np.array([1.0])
+    Te = 10 * u.eV
+    Ti = np.array([10]) * u.eV
+    ion_species = ["C-12 5+"]
+
+    alpha, Skw = thomson.spectral_density(
+        wavelengths,
+        probe_wavelength,
+        ne,
+        Te,
+        Ti,
+        fract=fract,
+        ion_species=ion_species,
+        probe_vec=probe_vec,
+        scatter_vec=scatter_vec,
+    )
+
+    return alpha, wavelengths, Skw
+
+
+def gen_multiple_ion_species_spectrum():
+    """
+    Generates an example Thomson scattering spectrum for multiple ion species
+    that also have drift velocities. Parameters are set to be in the
+    collective regime where ion species are important.
+    """
+    wavelengths = np.arange(520, 545, 0.01) * u.nm
+    probe_wavelength = 532 * u.nm
+    ne = 5e17 * u.cm ** -3
+    probe_vec = np.array([1, 0, 0])
+    scatter_vec = np.array([0, 1, 0])
+    fract = np.array([0.7, 0.3])
+    Te = 10 * u.eV
+    Ti = np.array([5, 5]) * u.eV
+    fluid_vel = np.array([300, 0, 0]) * u.km / u.s
+    ion_vel = np.array([[-500, 0, 0], [0, 500, 0]]) * u.km / u.s
+
+    # Use this to also test inputing ion species as Particle objects
+    ion_species = [Particle("p+"), Particle("C-12 5+")]
+
+    alpha, Skw = thomson.spectral_density(
+        wavelengths,
+        probe_wavelength,
+        ne,
+        Te,
+        Ti,
+        fract=fract,
+        ion_species=ion_species,
+        probe_vec=probe_vec,
+        scatter_vec=scatter_vec,
+        fluid_vel=fluid_vel,
+        ion_vel=ion_vel,
+    )
+
+    return alpha, wavelengths, Skw
+
+
+def gen_non_collective_spectrum():
+    """
+    Generates an example Thomson scattering spectrum in the non-collective
+    regime
+    """
+    wavelengths = np.arange(500, 570, 0.01) * u.nm
+    probe_wavelength = 532 * u.nm
+    ne = 5e15 * u.cm ** -3
+    probe_vec = np.array([1, 0, 0])
+    scatter_vec = np.array([0, 1, 0])
+    fract = np.array([1.0])
+    Te = 100 * u.eV
+    Ti = np.array([10]) * u.eV
+    ion_species = ["H+"]
+
+    alpha, Skw = thomson.spectral_density(
+        wavelengths,
+        probe_wavelength,
+        ne,
+        Te,
+        Ti,
+        fract=fract,
+        ion_species=ion_species,
+        probe_vec=probe_vec,
+        scatter_vec=scatter_vec,
+    )
+
+    return alpha, wavelengths, Skw
+
+
+def test_different_input_types():
+
+    # Define some constants
+    wavelengths = np.arange(520, 545, 0.01) * u.nm
+    probe_wavelength = 532 * u.nm
+    ne = 5e17 * u.cm ** -3
+    probe_vec = np.array([1, 0, 0])
+    scatter_vec = np.array([0, 1, 0])
+    fract = np.array([1.0])
+    Te = 10 * u.eV
+    Ti = np.array([10]) * u.eV
+    ion_species = ["C-12 5+"]
```

Comment:
```suggestion
    ion_species = "C-12 5+"
```

This will allow the tests to hit the following if-clause...
https://github.com/PlasmaPy/PlasmaPy/blob/b9552f4a5f1da5178b9100c12824664a53d15c7e/plasmapy/diagnostics/thomson.py#L155-L156